File: src/main/java/it/alian/gun/mesmerize/compat/Compat.java
package it.alian.gun.mesmerize.compat;

import org.bukkit.Bukkit;
import org.bukkit.World;
import org.bukkit.entity.Entity;
import org.bukkit.entity.LivingEntity;
import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Compat {

    public static List<Player> getOnlinePlayers() {
        List<Player> list = new ArrayList<>();
        for (World world : Bukkit.getWorlds()) {
            list.addAll(world.getPlayers());
        }
        return Collections.unmodifiableList(list);
    }

    public static Entity getByEntityId(int id, World world) {
        for (LivingEntity livingEntity : world.getLivingEntities()) {
            if (livingEntity.getEntityId() == id) return livingEntity;
        }
        return null;
    }

}

File: src/main/scala/it/alian/gun/mesmerize/compat/Compat.scala
package it.alian.gun.mesmerize.compat

import org.bukkit.entity.{Entity, Player}
import org.bukkit.{Bukkit, World}

import scala.collection.JavaConverters._

object Compat {

  def getOnlinePlayers: Seq[Player] =
    for {
      world <- Bukkit.getWorlds.asScala
      player <- world.getPlayers.asScala
    } yield player

  def getByEntityId(id: Int, world: World): Entity =
    (for (entity <- world.getLivingEntities.asScala if entity.getEntityId == id
    ) yield entity).headOption.orNull

}